src/features/twitter/controllers/twitter-controller.ts

```diff
@@ -20,20 +16,19 @@ export class TwitterController implements Controller {
   };
 
   login: AsyncRequestHandler = (req, res) => {
-    const authorization = req.headers.authorization;
-
-    if (!authorization) {
-      return res.status(401).json({ message: 'Unauthorized' });
+    try {
+      const url = this.loginTwitter.execute({
+        authorization: req.headers.authorization,
+      });
+
+      return res.json(url);
+    } catch (err) {
+      return res.status(401).json({ message: (err as Error).message });
     }
```

Comment:
vc vai ter que chamar o next aq, e dentro do service você joga o erro de que não tá autorizado
```suggestion
    } catch (error) {
      next(error)
    }
```